### DISCUSSION TITLE
How to set proxy for chatgpt-reborn?

### DISCUSSION THREAD

**Discussion by papandadj** [Christopher-Hayes/vscode-chatgpt-reborn#11] [Q&A]
### Describe the feature

How to set http proxy for chatgpt-reborn?

**Comment by i207M** [comment]
You could use Cloudflare to proxy OpenAI's API, and change the `baseurl` in the extension's settings.

```
const TELEGRAPH_URL = 'https://api.openai.com';

addEventListener('fetch', event => {
  event.respondWith(handleRequest(event.request))
})

async function handleRequest(request) {
  const url = new URL(request.url);
  url.host = TELEGRAPH_URL.replace(/^https?:\/\//, '');
  const modifiedRequest = new Request(url.toString(), {
    headers: request.headers,
    method: request.method,
    body: request.body,
    redirect: 'follow'
  });
  const response = await fetch(modifiedRequest);
  const modifiedResponse = new Response(response.body, response);
  modifiedResponse.headers.set('Access-Control-Allow-Origin', '*');
  return modifiedResponse;
}
```

**Comment by papandadj** [comment]
Thank you for your reply. I configure http.proxy in vscode. And it worked
<img width="382" alt="image" src="https://user-images.githubusercontent.com/25424898/226298576-aa0a9536-6a70-4c2a-afb0-b536d4e1749a.png">

**Reply by Christopher-Hayes** [reply]
Glad you got it working, I've migrated this to a discussion since it better suits this type of question.

As @i207M mentioned, the extension allows you to set the "Api Base Url", but if there's still a specific feature you would like added, feel free to create a new issue.
